### PR TITLE
neko: upstream Linux fix

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
This needs a full CI run as we are modifying some build code on Mac too.

File: Formula/neko.rb [hunk 1]
```diff
@@ -24,6 +24,16 @@ class Neko < Formula
...
   uses_from_macos "zlib"
 
+  on_linux do
+    depends_on "apr"
+    depends_on "apr-util"
+    depends_on "httpd"
+    # On mac, neko uses carbon. On Linux it uses gtk2
+    depends_on "gtk+"
+    depends_on "pango"
+    depends_on "sqlite"
```
Comment (iMichka) [inline Formula/neko.rb:34]: I think this should be `uses_from_macos "sqlite"`

Comment (danielnachun) [reply Formula/neko.rb]: Actually `depend_on "sqlite"` wasn't even needed, I missed that `uses_from_macos "sqlite"` was already in the formula.

**Comment by danielnachun** [comment]
I see that the failure here is identical to what I saw when I tested this in Linux CI, so this is definitely a problem with upstream CmakeLists.txt.  We'll probably have to open an issue, but I'll also see if there is an obvious patch we can make for now.

**Comment by danielnachun** [comment]
This is a bit complicated, but essentially the problem is in `mariadb-connector-c`, which `neko` installs its own copy of.  The CMake file has been fixed in `mariadb-connector-c`, but they haven't tagged a new release.  As soon as this happens, everything should work again.

**Comment by danielnachun** [comment]
The upstream issues for this are fixed.  Can I get a review on this?

File: Formula/neko.rb [hunk 2]
```diff
@@ -69,6 +70,15 @@ class Neko < Formula
   end
 
   def install
+    args = std_cmake_args
```
Comment (iMichka) [inline Formula/neko.rb:73]: This code is already present below.

**Pull request closed without merging**